deployment/node-feature-discovery/values.yaml

```diff
@@ -0,0 +1,207 @@
+    #        - "BMI2"
+    #        - "CLMUL"
+    #        - "CMOV"
+    #        - "CX16"
+    #        - "ERMS"
+    #        - "F16C"
+    #        - "HTT"
+    #        - "LZCNT"
+    #        - "MMX"
+    #        - "MMXEXT"
+    #        - "NX"
+    #        - "POPCNT"
+    #        - "RDRAND"
+    #        - "RDSEED"
+    #        - "RDTSCP"
+    #        - "SGX"
+    #        - "SSE"
+    #        - "SSE2"
+    #        - "SSE3"
+    #        - "SSE4.1"
+    #        - "SSE4.2"
+    #        - "SSSE3"
+    #      attributeWhitelist:
+    #  kernel:
+    #    kconfigFile: "/path/to/kconfig"
+    #    configOpts:
+    #      - "NO_HZ"
+    #      - "X86"
+    #      - "DMI"
+    #  pci:
+    #    deviceClassWhitelist:
+    #      - "0200"
+    #      - "03"
+    #      - "12"
+    #    deviceLabelFields:
+    #      - "class"
+    #      - "vendor"
+    #      - "device"
+    #      - "subsystem_vendor"
+    #      - "subsystem_device"
+    #  usb:
+    #    deviceClassWhitelist:
+    #      - "0e"
+    #      - "ef"
+    #      - "fe"
+    #      - "ff"
+    #    deviceLabelFields:
+    #      - "class"
+    #      - "vendor"
+    #      - "device"
+    #  custom:
+    #    - name: "my.kernel.feature"
+    #      matchOn:
+    #        - loadedKMod: ["example_kmod1", "example_kmod2"]
+    #    - name: "my.pci.feature"
+    #      matchOn:
+    #        - pciId:
+    #            class: ["0200"]
+    #            vendor: ["15b3"]
+    #            device: ["1014", "1017"]
+    #        - pciId :
+    #            vendor: ["8086"]
+    #            device: ["1000", "1100"]
+    #    - name: "my.usb.feature"
+    #      matchOn:
+    #        - usbId:
+    #          class: ["ff"]
+    #          vendor: ["03e7"]
+    #          device: ["2485"]
+    #        - usbId:
+    #          class: ["fe"]
+    #          vendor: ["1a6e"]
+    #          device: ["089a"]
+    #    - name: "my.combined.feature"
+    #      matchOn:
+    #        - pciId:
+    #            vendor: ["15b3"]
+    #            device: ["1014", "1017"]
+    #          loadedKMod : ["vendor_kmod1", "vendor_kmod2"]
+
+  podSecurityContext: {}
+    # fsGroup: 2000
+
+  securityContext: {}
+    # capabilities:
+    #   drop:
+    #   - ALL
+    # readOnlyRootFilesystem: true
+    # runAsNonRoot: true
+    # runAsUser: 1000
+
+  resources: {}
+    # We usually recommend not to specify default resources and to leave this as a conscious
+    # choice for the user. This also increases chances charts run on environments with little
+    # resources, such as Minikube. If you do want to specify resources, uncomment the following
+    # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+    # limits:
+    #   cpu: 100m
+    #   memory: 128Mi
+    # requests:
+    #   cpu: 100m
+    #   memory: 128Mi
+
+  nodeSelector: {}
+
+  tolerations:
+  - key: "node-role.kubernetes.io/master"
+    operator: "Equal"
+    value: ""
+    effect: "NoSchedule"
```

Comment:
I'm not sure if we should run nfd-worker on dedicated master nodes, by default. WDYT?

Reply:
maybe you want to look at https://github.com/kubernetes-sigs/node-feature-discovery-operator/blob/master/assets/worker/0700_worker_daemonset.yaml#L16 to set this rules on how nfd-worker will be placed

Reply:
I can find this beneficial for deploying in a development cluster. real deployment wouldn't need nfd-worker running on the master IMO, so thinking to set the defaults as to not run on master and give the user an option to override WDYT?

@ArangoGutierrez  is the node-feature-discovery operator intended to deploy NFD via OLM ?

Reply:
NFD can be deployed on OpenShift via OperatorHub or at any Kubernetes cluster via the Operator-framework community operators.
Notice that recently we added support for nfd-workers to run on master nodes, by request of the developers working on kubernetes edge support, this will be single node clusters, so some times, NFD master and worker would end up running on the same node. 
Upstream operator -> https://github.com/operator-framework/community-operators/tree/master/upstream-community-operators/nfd-operator
OpenShift-community operator -> https://github.com/operator-framework/community-operators/tree/master/community-operators/openshift-nfd-operator

Reply:
> IMO, so thinking to set the defaults as to not run on master and give the user an option to override WDYT?

That would sound good to me. To keep the two deployment methods (raw templates and helm) similar in this aspect. If we want to change the default behavior that should be done in both, then, I think.